### PR TITLE
Add signal based OTLP configuration

### PR DESCRIPTION
**Pull request by RassK**
## Why

Adds signal based OTLP exporter configuration variables.

## SDK

Related issue:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/5429

> Note: This is NOT currently utilized meaning AddOtlpExporter extension does NOT gain the ability to use these as of this PR.

* SDK's logic is using whole block of variables, there is no priority based logic.
* Logic is not fully finalized yet.

## What

Signal specified variables have higher priority than default variables. Auto instrumentation is using more detailed variables and overwrites options, default variable handling is left to SDK. 

Full list of supported variables:

```
OTEL_EXPORTER_OTLP_ENDPOINT
OTEL_EXPORTER_OTLP_HEADERS
OTEL_EXPORTER_OTLP_TIMEOUT
OTEL_EXPORTER_OTLP_PROTOCOL

OTEL_EXPORTER_OTLP_LOGS_ENDPOINT
OTEL_EXPORTER_OTLP_LOGS_HEADERS
OTEL_EXPORTER_OTLP_LOGS_TIMEOUT
OTEL_EXPORTER_OTLP_LOGS_PROTOCOL

OTEL_EXPORTER_OTLP_METRICS_ENDPOINT
OTEL_EXPORTER_OTLP_METRICS_HEADERS
OTEL_EXPORTER_OTLP_METRICS_TIMEOUT
OTEL_EXPORTER_OTLP_METRICS_PROTOCOL

OTEL_EXPORTER_OTLP_TRACES_ENDPOINT
OTEL_EXPORTER_OTLP_TRACES_HEADERS
OTEL_EXPORTER_OTLP_TRACES_TIMEOUT
OTEL_EXPORTER_OTLP_TRACES_PROTOCOL
```

## Tests

Extended some existing ones.

## Checklist

- [x] `CHANGELOG.md` is updated.
- [x] Documentation is updated.
- [x] New features are covered by tests.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ This component adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.h
 ### Added
 
 - Support for Operating System resource detector.
+- Support signal specific OTLP exporter variables. See list of supported variables
+  [here](https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/).
```
Comment (RassK) [inline CHANGELOG.md:14]: or either change to our internal doc url?

Comment (Kielek) [reply CHANGELOG.md:14]: Usually we are listing all new variables, without any links in changelog. If needed, I would put link to our doc.

**Review by nrcventura — APPROVED** [review]

**Comment by Kielek** [comment]
@RassK, could you please resolve confilcts?

File: src/OpenTelemetry.AutoInstrumentation/Configurations/Otlp/OtlpSettings.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    private static OtlpExportProtocol? GetExporterOtlpProtocol(OtlpSignalType signalType, Configuration configuration)
+    {
+        // the default in SDK is grpc. http/protobuf should be default for our purposes
+        var priorityVar = OtlpSpecConfigDefinitions.GetProtocolEnvVar(signalType);
+        var exporterOtlpProtocol = configuration.GetString(priorityVar) ??
+            configuration.GetString(OtlpSpecConfigDefinitions.DefaultProtocolEnvVarName);
+
+        if (string.IsNullOrEmpty(exporterOtlpProtocol))
```
Comment (zacharycmontoya) [inline src/OpenTelemetry.AutoInstrumentation/Configurations/Otlp/OtlpSettings.cs:80]: I realize that the logic in this method is unchanged from before (except for the signal-specific override), but how does this work? It seems like the configuration value is not used at all

Comment (zacharycmontoya) [reply src/OpenTelemetry.AutoInstrumentation/Configurations/Otlp/OtlpSettings.cs:80]: Nevermind, I reviewed the unit tests and this is completely expected and the env variable will be read in by the SDK itself 👍🏼

**Review by Kielek — APPROVED** [review]

**Pull request merged**